astropy/modeling/parameters.py

```diff
@@ -728,22 +728,22 @@ def __rtruediv__(self, val):
         return val / self.value
 
     def __eq__(self, val):
-        return (np.asarray(self) == np.asarray(val)).all()
+        return np.asanyarray(self) == np.asanyarray(val)
```

Comment:
> Note that asanyarray will just call the __array__ method and thus return a plain array!

That was the intent, I think.  It's not supposed to return a `Parameter` object or anything like that (though I will want it to return a `Quantity` for parameters with units).  Currently in #3852 the `.value` attribute just returns the raw value, and not the `Quantity`.  So I still need to use `np.asanyarray`.  However, I did not fix things yet so that `__array__` can return a `Quantity`.  Wouldn't that be good enough?